core/sql/langman/LmRoutineJavaObj.cpp

```diff
@@ -192,6 +192,7 @@ LmResult LmRoutineJavaObj::invokeRoutineMethod(
           (*emitRowPtr_)(NULL,0,&qs);
         }
     }
+  jni->DeleteLocalRef(jniResult);
```

Comment:
Sorry for not noticing this. It does seem dangerous to use data members from deleted objects. I will work on coding a fix and push another commit for this PR. Previously we know for certain that the ReturnInfo object on the Java side was never GCed. It was "live" always. So we were not exposed to this problem.